Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm

```diff
@@ -4624,53 +4624,64 @@ - (void)requestAutocorrectionRectsForString:(NSString *)input withCompletionHand
     });
 }
 
-- (void)requestRectsToEvadeForSelectionCommandsWithCompletionHandler:(void(^)(NSArray<NSValue *> *rects))completionHandler
+#if HAVE(UI_EDIT_MENU_INTERACTION)
+
+- (void)requestPreferredArrowDirectionForEditMenuWithCompletionHandler:(void(^)(UIEditMenuArrowDirection))completion
 {
-    if (!completionHandler) {
-        [NSException raise:NSInvalidArgumentException format:@"Expected a nonnull completion handler in %s.", __PRETTY_FUNCTION__];
-        return;
-    }
+    [self _requestEvasionRectsAroundSelectionIfNeeded:[completion = makeBlockPtr(completion)](const Vector<WebCore::FloatRect>& rects) {
+        completion(rects.isEmpty() ? UIEditMenuArrowDirectionAutomatic : UIEditMenuArrowDirectionUp);
```

Comment:
I'm assuming in the case where `rects` in non-empty, we use `UIEditMenuArrowDirectionUp` since the default is down, and `_requestEvasionRectsAroundSelectionIfNeeded` only looks _above_ the selection?

If that's the case, I think it would be nice to have a comment (or updated commit message) explaining why it's okay to always use `Up` in the evasion rect scenario. If my assumption is incorrect, then I would question why we never return  `UIEditMenuArrowDirectionDown`.

Reply:
Yes, that's correct! We use `UIEditMenuArrowDirectionUp ` to make the callout bar appear below the selection, in the case where we hit-test to (what is likely) clickable controls above the selection. If we don't detect any potential for overlap, we just go with `Automatic`, which (by default) means the arrow points down.

To make this a bit clearer, I'll explain this in the commit message, and also rename `_requestEvasionRectsAroundSelectionIfNeeded:` to `_requestEvasionRectsAboveSelectionIfNeeded:`, to make this a bit less mysterious.

I think in the future, we could consider hit-testing _below_ the selection as well, and in the case where there are only interactable controls below the selection, return `UIEditMenuArrowDirectionDown` to force the callout bar to point down (but for now, this just preserves the existing behavior, since we know the callout bar will always prefer pointing down by default).